### ISSUE TITLE
Migrate Release from Jenkins to Travis

### ISSUE THREAD

**Issue by benjchristensen** [ReactiveX/RxJava#1810]
I'd like to migrate our Gradle plugin from needing a Jenkins job to release to using Travis to do it.

We have proven the model works in the RxScala project using sbt and have a test project showing the hooks: https://github.com/ReactiveX/BuildInfrastructure/blob/master/.travis.yml#L5

The code for the grade plugin is at https://github.com/nebula-plugins/gradle-rxjava-project-plugin

The requirement is to modify the gradle plugin so we can pass in all necessary credentials either as environment variables or as command-line args for the bintray publishing.

This will be used for all the RxJava projects except RxScala which is using sbt.

**Comment by benjchristensen** [comment]
With the upgrade to Gradle 2 and Nebula RxJava Plugin 2.x this is almost working. 

The experiment can be seen at https://github.com/ReactiveX/BuildInfrastructure/blob/master/.travis.yml#L6

The outstanding issue is that it is not getting the right version number from the git tag. It is still resolving the version automatically.

**Comment by benjchristensen** [comment]
Also, there are some settings that we may want to change the defaults for: https://github.com/ReactiveX/BuildInfrastructure/blob/master/build.gradle#L23

```groovy
nebulaRelease {
    addReleaseBranchPattern(/\d+\.\d+\.\d+/)
    addReleaseBranchPattern('HEAD')
}

if (project.hasProperty('release.useLastTag')) {
    tasks.prepare.enabled = false
}
```

**Comment by benjchristensen** [comment]
Releasing via Travis now works. Version 1.0.5 was built and released to Bintray solely by adding the tag `v1.0.5`. 

The log showing the release can be seen here: https://travis-ci.org/ReactiveX/RxJava/builds/49421528#L3313

Thank you @rspieldenner and @quidryan for enabling this!
